labgrid/util/agents/sysfsgpio.py

```diff
@@ -0,0 +1,84 @@
+"""
+This module implements implements switching GPIOs via sysfs GPIO kernel interface.
+
+Takes an integer property 'index' which refers to the already exported GPIO device.
+
+"""
+import logging
+import os
+
+class GpioDigitalOutput:
+    _gpio_sysfs_path_prefix = '/sys/class/gpio'
+    _buffered_file_access=False
+
+    def _assert_gpio_line_is_exported(index):
+        gpio_sysfs_path = os.path.join(GpioDigitalOutput._gpio_sysfs_path_prefix, 'gpio{0}'.format(index))
+        if not os.path.exists(gpio_sysfs_path):
+            export_sysfs_path = os.path.join(GpioDigitalOutput._gpio_sysfs_path_prefix, 'export')
+            with open(export_sysfs_path, mode = 'r+', buffering = GpioDigitalOutput._buffered_file_access, closefd = True) as export:
+                export.write(str(index))
+        if not os.path.exists(gpio_sysfs_path):
+            raise ValueError("Device not found")
+
+    def __init__(self, **kwargs):
+        index = kwargs['index']
+        self._logger = logging.getLogger("Device: ")
+        GpioDigitalOutput._assert_gpio_line_is_exported(index)
+        gpio_sysfs_path = os.path.join(GpioDigitalOutput._gpio_sysfs_path_prefix, 'gpio{0}'.format(index))
+        gpio_sysfs_direction_path = os.path.join(gpio_sysfs_path, 'direction')
+        direction_fd = os.open(gpio_sysfs_direction_path, flags=os.O_WRONLY)
```

Comment:
this could be
```
self._logger.debug("Configuring GPIO {idx} as output.".format(idx = index))
with open(gpio_sysfs_direction_path, 'wb') as direction_fd:
    direction_fd.write(b'out')
```

Reply:
db91e47da5283d5b8556cc062e86c01d0d271f05